Source/core/CyclicBuffer.cpp

```diff
@@ -366,7 +366,7 @@ namespace Core {
                 break;
             }
 
-            Cursor cursor(*this, oldTail, length);
+            Cursor cursor(*this, oldTail, std::min(length, result));
```

Comment:
I am hoping that this is not needed. Length should always be oke. It could be that after line 363 more data was written into the cyclic buffer and it is up to the consumer of the cursor (GetReadSize() in line 370) to see if he want to read it. The only restriction for this Consumer is the amount of data that can be stored into the passed in Read Buffer with length " length". So I do not think this change fixes any usecase, potentially it only slows down the read process casue now if during line 364 and 370 more data is added to the cyclic buffer it is not treated as being available, where the consumer of the cursor in line 370 moight now conclude that it can not read that data as the buffer to store it in is to small (limited to the amount of data that was available at line 363) but maybe the read required 2 or more writes for a full frame to load.
Is there a uses case that is now fixed with this change? If so I would like to evaluate that use-case.

Reply:
You might have to clarify, perhaps, offline. I believe I shared the reasoning with @MFransen69 via our internal communication medium.

Reply:
Hi @msieben I cannot remember we discussed this in too much detail, so better indeed to share it here.

Reply:
@MFransen69 I already shared the transcript with @pwielders

Reply:
If analysis is correct, but correct me if I am wrong.

In short what has been shared.

https://github.com/rdkcentral/Thunder/blob/master/Source/core/CyclicBuffer.cpp#L369-L370
https://github.com/rdkcentral/Thunder/blob/master/Source/core/CyclicBuffer.cpp#L694
https://github.com/rdkcentral/Thunder/blob/master/Source/core/CyclicBuffer.h#L57
https://github.com/rdkcentral/Thunder/blob/master/Source/core/CyclicBuffer.h#L80

Suggest result is equivalent to length.

https://github.com/rdkcentral/Thunder/blob/master/Source/core/CyclicBuffer.cpp#L385

Under the  assumption of available data bufferLength becomes equivalent to length.

This may affect memcpy if buffer size allows and then length 'dictates' what will be copied.

https://github.com/rdkcentral/Thunder/blob/master/Source/core/CyclicBuffer.cpp#L390
https://github.com/rdkcentral/Thunder/blob/master/Source/core/CyclicBuffer.cpp#L407
https://github.com/rdkcentral/Thunder/blob/master/Source/core/CyclicBuffer.cpp#L410

Reply:
@pwielder, if analysis is correct and performance is paramount then an ASSERT might be used instead since the std::min is equivalent to a comparison and assignment, and, the comparison suffices.